### PR TITLE
Updated Transpile CLI to use Workspace config

### PR DESCRIPTION
**Pull request by vijaypavann-db**
Problem:
The existing `Transpile CLI command` expects all the flags such as `source`, `input-sql`, `output-folder` to be passed as part of the CLI Invocation. But, we already captured these values in `config.yml` as part of the `remorph` Installation process.

Solution:
This PR updates the `Transpile CLI command` to use the Databricks workspace `config.yml` to run the Transpile process.

Steps:
* Load the `config.yml` as `MorphConfig`
* Validate the attributes
* Run the `Transpile` process

### CODE REVIEW

File: src/databricks/labs/remorph/helpers/transpile_utils.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
+import json
+import logging
+import os
+
+from databricks.sdk import WorkspaceClient
```
Comment (sundarshankar89) [inline src/databricks/labs/remorph/helpers/transpile_utils.py:5]: move this to context folder have them as part of transpile context. we will later refactor them using application context.

Comment (sundarshankar89) [reply src/databricks/labs/remorph/helpers/transpile_utils.py:5]: cc: @bishwajit-db thoughts?

**Review by sundarshankar89 — CHANGES_REQUESTED** [review]

File: src/databricks/labs/remorph/helpers/transpile_utils.py (hunk 1 above)
Comment (bishwajit-db) [reply src/databricks/labs/remorph/helpers/transpile_utils.py:5]: makes sense

Comment (vijaypavann-db) [reply src/databricks/labs/remorph/helpers/transpile_utils.py]: Done, please check

File: src/databricks/labs/remorph/cli.py [hunk 2]
```diff
@@ -7,72 +6,26 @@
...
+def transpile(w: WorkspaceClient):
+    """[EXPERIMENTAL] reconciles source to databricks datasets"""
+    logger.info(f"user: {w.current_user.me()}")
 
-    status = morph(w, config)
+    installation = Installation.assume_user_home(w, "remorph")
 
-    print(json.dumps(status))
+    context = TranspileContext(w, installation)
+    context.run()
```
Comment (nfx) [inline src/databricks/labs/remorph/cli.py:28]:
```suggestion
    ctx = TranspileContext(w)
    ctx.run()
```
create Installation instance from transpile context

File: src/databricks/labs/remorph/context/transpile_context.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+    def _load_config(self) -> MorphConfig | None:
+        transpile_config = None
+        try:
+            logger.info("Loading MorphConfig `config.yml` from Databricks Workspace...")
+            transpile_config = self._installation.load(MorphConfig)
+        except NotFound as err:
+            logger.warning(f"Cannot find previous `transpile` installation: {err}")
+        except (PermissionDenied, SerdeError, ValueError, AttributeError):
+            logger.warning(f"Existing installation at {self._installation.install_folder()} is corrupted. Skipping...")
```
Comment (nfx) [inline src/databricks/labs/remorph/context/transpile_context.py:39]:
```suggestion
    def __init__(self, w: WorkspaceClient):
        self._ws = w
    
    def replace(self, **kwargs):
        """Replace cached properties for unit testing purposes."""
        for key, value in kwargs.items():
            self.__dict__[key] = value
        return self

    @cached_property
    def workspace_client(self) -> WorkspaceClient:
        raise self._ws
    
    @cached_property
    def installation(self) -> Installation:
        raise Installation.assume_user_home(self._workspace_client, "remorph")

    @cached_property
    def prompts(self) -> Prompts:
        raise Prompts()

    @cached_property
    def morph_config(self) -> MorphConfig:
        try:
            logger.debug("Loading MorphConfig `config.yml` from Databricks Workspace...")
            return self.installation.load(MorphConfig)
        except NotFound as err:
            raise NotFound("Cannot find previous `transpile` installation") from err
        except (PermissionDenied, SerdeError, ValueError, AttributeError):
            raise RuntimeWarning(f"Existing installation at {self.installation.install_folder()} is corrupted. Skipping...")
```

Rewrite `Context` classes to only be creating other objects. See how we do it in UCX https://github.com/databrickslabs/ucx/blob/main/src/databricks/labs/ucx/contexts/application.py

**Review by nfx — CHANGES_REQUESTED** [review]

File: src/databricks/labs/remorph/cli.py (hunk 2 above)
Comment (vijaypavann-db) [reply src/databricks/labs/remorph/cli.py]: Will refactor to the latest Application context

File: src/databricks/labs/remorph/context/transpile_context.py (hunk 3 above)
Comment (vijaypavann-db) [reply src/databricks/labs/remorph/context/transpile_context.py:39]: Will refactor to the latest Application context

**Comment by nfx** [comment]
closing this pr as stale

**Pull request closed without merging**